crates/compiler/can/src/desugar.rs

```diff
@@ -213,62 +213,132 @@ fn new_op_call_expr<'a>(
             let left = desugar_expr(env, scope, left);
             let right = desugar_expr(env, scope, right);
 
-            let mut branches = Vec::with_capacity_in(2, env.arena);
-            let mut branch_1_patts = Vec::with_capacity_in(1, env.arena);
-            let mut branch_1_patts_args = Vec::with_capacity_in(1, env.arena);
-            let success_var = env.arena.alloc_str(
-                format!(
-                    "success_BRANCH1_{}_{}",
-                    left.region.start().offset,
-                    left.region.end().offset
-                )
-                .as_str(),
-            );
-            branch_1_patts_args.push(Loc::at(
-                left.region,
-                Pattern::Identifier { ident: success_var },
+            let ok_var = env.arena.alloc_str(&format!(
+                "double_question_ok_{}_{}",
```

Comment:
This might conflict with other names in the outer scope. i would suggest prefixing with a unique symbol not accepted in identifiers like @

Reply:
Prefixed the vars with # as is standard, good catch